tests/tests_pytorch/core/test_saving.py

```diff
@@ -116,3 +116,17 @@ def test_load_from_checkpoint_warn_on_empty_state_dict(tmp_path):
     with pytest.warns(UserWarning, match="contains no parameters"):
         model = BoringModel.load_from_checkpoint(tmp_path / "checkpoint.ckpt", strict=False)
     assert model.device.type == "cpu"
+
+
+def test_load_from_checkpoint_classmethod(tmp_path):
```

Comment:
Can you instead test the restricted checkpoint by creating a class with a classmethod that sets it? We don't need to create a checkpoint and load a model to test it